yt/data_objects/data_containers.py

```diff
@@ -836,6 +849,19 @@ def create_firefly_object(
 
             ## explicitly go after the fields we want
             for field, units in zip(fields_to_include, fields_units):
+                ## Only interested in fields with the current particle type,
+                ## whether that means general fields or field tuples
+                if isinstance(field, tuple):
+                    ftype, fname = field
+                    if not (ftype == ptype or ftype == "all"):
+                        continue
+                    if fname not in kysd[ptype]:
+                        raise YTFieldNotFound(field=(ftype, fname), ds=self.ds)
+                    field = fname
+                elif field not in kysd[ptype]:
+                    mylog.warning("requested (but not available) %s %s", ptype, field)
+                    continue
```

Comment:
with my main suggestion to use `_determine_fields` uphill, this would simplify as
```suggestion
                ftype, fname = field
                if ftype not in (ptype, "all"):
                    continue
                field = fname
```
while we're here, could you maybe also avoid changing the type of the `field` variable from `tuple[str, str]` to `str`, and just use `fname` where appropriate instead ?

Reply:
Fixed the `field` conversion ea4eab5. I'm not sure why I didn't do it this way earlier.